hydromt/components/vector.py

```diff
@@ -250,164 +263,63 @@ def write_vector(
-    def read(
-        self,
-        components: List = None,
-    ) -> None:
-        """Read the complete model from model files.
-
-        Parameters
-        ----------
-        components : List, optional
-            List of model components to read, each should have an
-            associated read_<component> method.
-            By default ['config', 'maps', 'vector', 'geoms', 'tables',
-            'forcing', 'states', 'results']
-        """
-        components = components or [
-            "config",
-            "vector",
-            "geoms",
-            "tables",
-            "forcing",
-            "states",
-            "results",
-        ]
-        super().read(components=components)
-
-    def write(
-        self,
-        components: List = None,
-    ) -> None:
-        """Write the complete model schematization and configuration to model files.
-
-        Parameters
-        ----------
-        components : List, optional
-            List of model components to write, each should have an
-            associated write_<component> method. By default ['config',
-            'maps', 'vector', 'geoms', 'tables', 'forcing', 'states']
-        """
-        components = components or [
-            "config",
-            "vector",
-            "geoms",
-            "tables",
-            "forcing",
-            "states",
-        ]
-        super().write(components=components)
-
-    def _test_equal(self, other, skip_component=None) -> Tuple[bool, Dict]:
-        """Test if two models including their data components are equal.
-
-        Parameters
-        ----------
-        other : Model (or subclass)
-            Model to compare against
-        skip_component: list
-            List of components to skip when testing equality. By default root.
-
-        Returns
-        -------
-        equal: bool
-            True if equal
-        errors: dict
-            Dictionary with errors per model component which is not equal
-        """
-        skip_component = skip_component or []
-        if "vector" not in skip_component:
-            # add vector to skip_component list
-            skip_component.append("vector")
-
-        equal, errors = super()._test_equal(other, skip_component=skip_component)
-
-        # test vector separately especially for the geometry
-        geods = self.vector
-        geods_other = other.vector
-        # test geometry
-        gdf = geods.vector.geometry.to_frame("geometry")
-        gdf_other = geods_other.vector.geometry.to_frame("geometry")
-        errors.update(_check_equal(gdf, gdf_other, name="vector.geometry"))
-        # test vector data only
-        if geods.vector.geom_format == "xy":
-            drop_vars = [geods.vector.x_name, geods.vector.y_name]
-        else:
-            drop_vars = [geods.vector.geom_name]
-        ds = geods.drop_vars(drop_vars)
-        ds_other = geods_other.drop_vars(drop_vars)
-        errors.update(_check_equal(ds, ds_other, name="vector.data"))
```

Comment:
This part was not transferred. Maybe each component should have its own test_equal function that can be called from Model.test_equal ? That decision would impact other components @Tjalling-dejong and @savente93

Reply:
So this depends on how we want to define equality. if it is just "has the same components with the right name and type" then model can just do this on it's own. if equality means "has the same data" then yet, each component will need to implement their own equality check. However, if we want this then this equality check will have to become a required part of the component API. is it important enough to us to for plugins to do this? (genuine question, no sarcasm intended). I am okay with this, but we should be aware and document properly that this could be a very expensive operation depending on how much data is involved.

Reply:
I was also wondering what this is used for. And it's mainly for unit tests to see if reading and writing is implemented correctly. For that reason I would opt for moving it to the test utilities, which I did (see test_equal.py)

Reply:
I've put this code back where it belongs.